### ISSUE TITLE
Need better text handling

### ISSUE THREAD

**Issue by Eyesonjune18** [tui-rs-revival/ratatui#162]
## Problem
The way that this crate handles text is flawed, in my opinion. There are two major pain points and one minor pain point that I have encountered when dealing with `Text`, `Spans`, and `Span`.
### Major pain points
- `Text` is an ambiguous term to describe "a collection of lines." `Spans` is a bad name because it creates confusion between `Spans` and `Span`s. This is especially an issue in code comments, where markdown's inline code blocks are not supported.
- `Spans` should not have an unnamed field. `Spans.0` makes code unreadable in a way that is entirely unnecessary; the reason I suspect this has not been changed is because it *is* a breaking change. But I believe it to be a warranted one. It is my opinion that no crate should have publicly-exposed unnamed fields.
### Minor pain points
- The fact that representing line breaks requires turning your `Span` into a `Spans` and placing it in a `Text` is rather annoying. I call this a minor pain point because, although it makes manipulating text (especially existing text) rather unwieldy, it may have a good reason for being implemented the way it is. Feedback on this would be appreciated.

## Solution
- `Spans` needs to be renamed to `Line`.
- `Spans`/`Line` needs a named field. If the previous suggestion is adopted, I would recommend `Line.spans`.
- It may be a good idea to rename `Text` to something else - though I'm not really aware of any specific name for a collection of lines.
- This will be a breaking change, so I think that general text-manipulation improvements should be bundled into a minor version release like `0.21.0`.

## Alternatives
I'm not sure what alternatives to this feature would be - feel free to let me know any ideas you have.

**Comment by joshka** [comment]
I agree - though I'd probably state this less as "flawed" and more that it lacks an ergonomic external API ;) My guess is that there is a large amount of historical reasons for the naming.

**Comment by Eyesonjune18** [comment]
I guess "unergonomic" is a slightly nicer way to put it :)
Though there may be historical reasons for the nomenclature, I do believe that we should change it regardless.

**Comment by Eyesonjune18** [comment]
https://github.com/tui-rs-revival/ratatui/pull/172

**Comment by joshka** [comment]
Most of the problems mentioned in this except Text have been fixed by introducing `Line`, and renaming Text seems more of a discussion that a specific problem, so I'm moving this over to discussions to help trim down the issues list.
